### PR TITLE
ss-266 Export the Pretty Stuff (#266)

### PR DESCRIPTION
**Pull request by ScottSchumacker**
1. I created a new function called generate_comm_data. This function pulls community usage metrics for a specific package. This function has been exported for use. This function returns a table of community usage numbers.
2. I cleaned up the insert_community_metrics_to_db function to call this new function generate_comm_data.
3. I exported the build_comm_plotly function. In addition, I added an optional parameter. The user can now use this function and type in any package name as an argument and a plotly data visualization is returned.
4. I updated the documentation for the functions mentioned above.
5. I updated the NEWS file as well as the DESCRIPTION. Added myself as a contributor and updated the version.

These updates are related to Issue #266

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -268,15 +268,26 @@ auto_font <- function(txt, txt_max = 45, size_min = .75, size_max = 1.5,
 
 
 #' The 'Build Community plot' function
+#' @description 
+#' build_comm_plotly() is responsible for building the plotly graphic of community usage metrics
```
Comment (AARON-CLARK) [inline R/utils.R:272]: This is kind of weak description, haha. Try to be more specific.

```suggestion
#' Responsible for building an interactive `{plotly}` graphic containing the trend line for number of CRAN pkg downloads by month.
```

File: R/utils.R [hunk 2]
```diff
@@ -268,15 +268,26 @@ auto_font <- function(txt, txt_max = 45, size_min = .75, size_max = 1.5,
...
 #' 
-#' @param data a data.frame
-#' 
+#' @param data a data.frame that is built in databaseView.R from get_comm_data()
```
Comment (AARON-CLARK) [inline R/utils.R:274]: You have to remember that this is an exported function, so this parameter description won't make any sense to our users. Try:

```suggestion
#' @param data a data.frame containing monthly download data, built using `generate_comm_data()`. This argument is optional, but if `NULL`, a `pkg_name` must be provided.
```

File: R/utils.R [hunk 3]
```diff
@@ -268,15 +268,26 @@ auto_font <- function(txt, txt_max = 45, size_min = .75, size_max = 1.5,
...
+#' @param pkg_name a string of a package name. This parameter is optional.
```
Comment (AARON-CLARK) [inline R/utils.R:275]:
```suggestion
#' @param pkg_name a string, containing a package name. This parameter is optional. If missing, you must provide a data.frame for the `data` argument.
```

File: R/utils.R [hunk 4]
```diff
@@ -268,15 +268,26 @@ auto_font <- function(txt, txt_max = 45, size_min = .75, size_max = 1.5,
...
+#' metricGraph <- build_comm_plotly(pkg_name = "ggplot2")
 #' @import dplyr
 #' @importFrom lubridate NA_Date_ interval
 #' @importFrom glue glue
 #' @importFrom plotly plot_ly layout add_segments add_annotations config
-#' 
-build_comm_plotly <- function(data) {
+#' @export
+build_comm_plotly <- function(data, pkg_name = NULL) {
+
```
Comment (AARON-CLARK) [inline R/utils.R:285]: So, the start of the function should contain some checks to make sure the user provided one of the `data` or `pkg_name` arguments. If not, you should return an error using `stop()`.

File: R/utils.R [hunk 5]
```diff
@@ -268,15 +268,26 @@ auto_font <- function(txt, txt_max = 45, size_min = .75, size_max = 1.5,
...
+#' @examples 
+#' metricGraph <- build_comm_plotly(pkg_name = "ggplot2")
 #' @import dplyr
 #' @importFrom lubridate NA_Date_ interval
 #' @importFrom glue glue
 #' @importFrom plotly plot_ly layout add_segments add_annotations config
-#' 
-build_comm_plotly <- function(data) {
+#' @export
+build_comm_plotly <- function(data, pkg_name = NULL) {
```
Comment (AARON-CLARK) [inline R/utils.R:284]: default value for data should probably be `NULL` as well for argument validation. See next comment.

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @ScottSchumacker, 

Great PR! This will be valuable for the new release. I have a few overall comments:

* the function `generate_comm_data()` should be located in `R/utils.R` near `get_latest_pkg_info()` since they are so related.

* Could you get this function to work? I kept getting a "No Community Usage Metrics available" message in the app. And `cranlogs::cran_downloads()` was giving me a 504 timeout message when running locally. This appears to be a problem on `dev`, `master`, and even our shinyapps.io deployment which is very old. I couldn't find any related open issues on {cranlogs} github though... Interested to hear your experience.

* Feel free to add a unit test for these functions, either in this PR or in the next immediately following a merge.

A few specific comments below:

**Comment by AARON-CLARK** [comment]
@ScottSchumacker, I was able to get `cranlogs::cran_downloads()` to work finally. I remembered that I had `fontawesome 0.4.0` installed for testing purposes. As such, I reverted back to v0.3.0 and the issue resolved itself. My apologies!

File: NEWS.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 # riskassessment (development version)
 
+* Added a new function called generate_comm_data(). I have also exported this function (#266).
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply NEWS.md:3]: @AARON-CLARK Updated

File: R/utils.R (hunk 5 above)
Comment (ScottSchumacker) [reply R/utils.R:284]: @AARON-CLARK Updated

File: R/utils.R (hunk 3 above)
Comment (ScottSchumacker) [reply R/utils.R:275]: @AARON-CLARK Updated

File: R/utils.R (hunk 2 above)
Comment (ScottSchumacker) [reply R/utils.R:274]: @AARON-CLARK Updated

File: R/utils.R (hunk 1 above)
Comment (ScottSchumacker) [reply R/utils.R:272]: @AARON-CLARK Updated

File: R/generate_comm_data.R [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+#' @importFrom cranlogs cran_downloads
+#' @importFrom lubridate year month
+#' @importFrom glue glue 
+#' @importFrom rvest read_html html_node html_table html_text
+#' @importFrom loggit loggit
+#' @importFrom stringr str_remove_all
+#' @importFrom tidyr tibble
+#' @examples 
+#' metricDF <- generate_comm_data("ggplot2")
+#' metricDF <- generate_comm_data("dplyr")
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply R/generate_comm_data.R:18]: @AARON-CLARK Updated

File: R/generate_comm_data.R [hunk 8]
```diff
@@ -0,0 +1,88 @@
+#' generate_comm_data()
+#' 
+#' @description 
+#' generate_comm_data() is a function that extracts usage metrics for a given package.
+#' @returns A tibble of package usage metrics
+#' @param pkg_name A string name of the package.
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply R/generate_comm_data.R:6]: @AARON-CLARK Updated

File: R/generate_comm_data.R [hunk 9]
```diff
@@ -0,0 +1,88 @@
+#' generate_comm_data()
+#' 
+#' @description 
+#' generate_comm_data() is a function that extracts usage metrics for a given package.
+#' @returns A tibble of package usage metrics
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply R/generate_comm_data.R:5]: @AARON-CLARK Updated

File: R/generate_comm_data.R [hunk 10]
```diff
@@ -0,0 +1,88 @@
+#' generate_comm_data()
+#' 
+#' @description 
+#' generate_comm_data() is a function that extracts usage metrics for a given package.
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply R/generate_comm_data.R:4]: @AARON-CLARK Updated

File: NEWS.md [hunk 11]
```diff
@@ -1,5 +1,8 @@
...
+* Updated and cleaned up the insert_community_metrics_to_db() function (#266).
```
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply NEWS.md:4]: @AARON-CLARK Updated

File: NEWS.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ScottSchumacker) [reply NEWS.md:3]: @AARON-CLARK Updated

**Comment by AARON-CLARK** [comment]
@ScottSchumacker, I see you're still working on this. Just a reminder: you'll want to re-run `run_dev.R` when finished with your changes to update the pkg `Rd` files. Also, you'll probably need to increment the package dev version again since `dev` has had a couple of merged PRs the last few days. Thanks!

**Comment by AARON-CLARK** [comment]
@ScottSchumacker, closing so you can [open a new PR](https://github.com/pharmaR/riskassessment/compare/dev...ss-266New) into `dev` from `ss-266New`.

**Pull request closed without merging**